cli/tests/rsa_signature.go

```diff
@@ -0,0 +1,409 @@
+	NetworkPrivateKey          string                           `yaml:"NetworkPrivateKey" env:"NETWORK_PRIVATE_KEY" env-description:"private key for network identity"`
+	WsAPIPort                  int                              `yaml:"WebSocketAPIPort" env:"WS_API_PORT" env-description:"Port to listen on for the websocket API."`
+	WithPing                   bool                             `yaml:"WithPing" env:"WITH_PING" env-description:"Whether to send websocket ping messages'"`
+	SSVAPIPort                 int                              `yaml:"SSVAPIPort" env:"SSV_API_PORT" env-description:"Port to listen on for the SSV API."`
+	LocalEventsPath            string                           `yaml:"LocalEventsPath" env:"EVENTS_PATH" env-description:"path to local events"`
+}
+
+var cfg config
+var globalArgs global_config.Args
+
+// RsaCmd is the command to rsa test
+var RsaCmd = &cobra.Command{
+	Use:   "test-rsa",
+	Short: "Test Rsa Wrong signature",
+	Run: func(cmd *cobra.Command, args []string) {
+		logger, err := setupGlobal()
+		ctx := context.Background()
+		if err != nil {
+			log.Fatal("could not create logger", err)
+		}
+
+		metricsReporter := metricsreporter.New(
+			metricsreporter.WithLogger(logger),
+		)
+
+		networkConfig, err := setupSSVNetwork(logger)
+		if err != nil {
+			logger.Fatal("could not setup network", zap.Error(err))
+		}
+		cfg.DBOptions.Ctx = ctx
+		db, err := setupDB(logger, networkConfig.Beacon.GetNetwork())
+		if err != nil {
+			logger.Fatal("could not setup db", zap.Error(err))
+		}
+
+		nodeStorage, operatorData := setupOperatorStorage(logger, db)
+
+		operatorKey, _, _ := nodeStorage.GetPrivateKey()
+		keyBytes := x509.MarshalPKCS1PrivateKey(operatorKey)
+		hashedKey, _ := rsaencryption.HashRsaKey(keyBytes)
+		keyManager, err := ekm.NewETHKeyManagerSigner(logger, db, networkConfig, cfg.SSVOptions.ValidatorOptions.BuilderProposals, hashedKey)
+		if err != nil {
+			logger.Fatal("could not create new eth-key-manager signer", zap.Error(err))
+		}
+
+		cfg.P2pNetworkConfig.Ctx = ctx
+
+		if err != nil {
+			logger.Fatal("could not connect to execution client", zap.Error(err))
+		}
+
+		var validatorCtrl validator.Controller
+		cfg.P2pNetworkConfig.Permissioned = func() bool {
+			return false
+		}
+
+		cfg.P2pNetworkConfig.NodeStorage = nodeStorage
+		cfg.P2pNetworkConfig.OperatorPubKeyHash = format.OperatorID(operatorData.PublicKey)
+		operatorId := 0
+		cfg.P2pNetworkConfig.OperatorID = func() spectypes.OperatorID {
+			operatorId++
+			return spectypes.OperatorID(operatorId)
+		}
+		cfg.P2pNetworkConfig.FullNode = cfg.SSVOptions.ValidatorOptions.FullNode
+		cfg.P2pNetworkConfig.Network = networkConfig
+
+		validatorsMap := validatorsmap.New(ctx)
+
+		dutyStore := dutystore.New()
+		cfg.SSVOptions.DutyStore = dutyStore
+
+		messageValidator := validation.NewMessageValidator(
```

Comment:
we don't really need to validate messages..

Reply:
the process of parsing messages includes decode and p2p requires messageValidator

Reply:
yes but you could create a small message validator that doesn't goes through all validation logic. not super critical but makes more sense here.